packages/google_maps_flutter/google_maps_flutter_web/lib/src/google_maps_controller.dart

```diff
@@ -413,6 +436,113 @@ class GoogleMapController {
     return _markersController?.isInfoWindowShown(markerId) ?? false;
   }
 
+  // Add My Location widget to right bottom
+  HtmlElement _createMyLocationButton() {
+    final HtmlElement controlDiv = DivElement();
+    controlDiv.style.marginRight = '10px';
+
+    final HtmlElement firstChild = ButtonElement();
+    firstChild.className = 'gm-control-active';
+    firstChild.style.backgroundColor = '#fff';
+    firstChild.style.border = 'none';
+    firstChild.style.outline = 'none';
+    firstChild.style.width = '40px';
+    firstChild.style.height = '40px';
+    firstChild.style.borderRadius = '2px';
+    firstChild.style.boxShadow = '0 1px 4px rgba(0,0,0,0.3)';
+    firstChild.style.cursor = 'pointer';
+    firstChild.style.padding = '8px';
+    controlDiv.append(firstChild);
+
+    final HtmlElement secondChild = DivElement();
+    secondChild.style.width = '24px';
+    secondChild.style.height = '24px';
+    secondChild.style.backgroundImage =
+        'url(https://maps.gstatic.com/tactile/mylocation/mylocation-sprite-2x.png)';
+    secondChild.style.backgroundSize = '240px 24px';
+    secondChild.style.backgroundPosition = '0px 0px';
+    secondChild.style.backgroundRepeat = 'no-repeat';
+    secondChild.id = 'you_location_img';
+    firstChild.append(secondChild);
+
+    firstChild.addEventListener('click', (_) {
+      String imgX = '0';
+      // Add animation when find current location
+      final Timer timer =
+          Timer.periodic(const Duration(milliseconds: 500), (_) {
+        imgX = (imgX == '-24') ? '0' : '-24';
+        document.getElementById('you_location_img')?.style.backgroundPosition =
+            '${imgX}px 0px';
```

Comment:
Also, this should be done with CSS, check out this demo:

https://jsfiddle.net/ditman/kqmu5Lp7

(The JS is there only to set/remove the "waiting" class in the `secondChild`, not to drive the animation!)